src/interaction/LennardJonesSoftcoreTI.hpp

```diff
@@ -0,0 +1,281 @@
+/*
+  Copyright (C) 2012,2013,2016
+      Max Planck Institute for Polymer Research
+  Copyright (C) 2008,2009,2010,2011
+      Max-Planck-Institute for Polymer Research & Fraunhofer SCAI
+  
+  This file is part of ESPResSo++.
+  
+  ESPResSo++ is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+
+// ESPP_CLASS
+#ifndef _INTERACTION_LENNARDJONESSOFTCORETI_HPP
+#define _INTERACTION_LENNARDJONESSOFTCORETI_HPP
+
+#include "FixedPairListInteractionTemplate.hpp"
+#include "FixedPairListTypesInteractionTemplate.hpp"
+#include "Potential.hpp"
+
+namespace espressopp {
+  namespace interaction {
+    /** This class provides methods to compute forces and energies of
+	the softcore Lennard Jones potential
+        in a Thermodynamic Integration simulation
+    **/
+    class LennardJonesSoftcoreTI : public PotentialTemplate< LennardJonesSoftcoreTI > {
+    private:
+      real epsilonA, epsilonB;
+      real ff1A, ff2A;
+      real ff1B, ff2B;
+      bool annihilate; //if true, atoms in pidsTI are annihilated, otherwise they are decoupled
+      std::vector<real> pidsTI; //PIDs of particles whose LJ interaction is zero in TI state B
```

Comment:
That's a mistake, should be `longint`. 

The length is typically N<100 particles but could in theory be much longer.

I just did some quick benchmarking for the worst-case scenario where the element is never found in the set/vector (which is what happens most of the time in typical applications of this class), looks like already when N>15 it's worth using set.